### PR TITLE
feat(persist-web-storage): add withCookie

### LINKED ISSUE #825: Add cookie adapter to reatom/persist-web-storage
@Laniman is already working on it

### PR DESCRIPTION
**Pull request by Laniman**
Closes #825

### CODE REVIEW

File: packages/persist-web-storage/src/reatomPersistCookie.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        const cookie = document.cookie
+
+        if (cookie === '') return null
+
+        const dataStr = cookie
+          .split('; ')
+          .find((row) => row.startsWith(`${key}=`))
+          ?.split('=')[1]
+
+        if (!dataStr) return null
```
Comment (artalar) [inline packages/persist-web-storage/src/reatomPersistCookie.ts]: Move it inside the `try`

File: packages/persist-web-storage/src/reatomPersistCookie.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+          }
+        } else {
+          rec.to = options.maxAge * 1000 + now
+        }
+
+        const memCache = ctx.get(memCacheAtom)
+        memCache.set(key, rec)
+
+        ctx.schedule(() => {
+          document.cookie = `${key}=${JSON.stringify(rec)}${stringifyAttrs(
```
Comment (artalar) [inline packages/persist-web-storage/src/reatomPersistCookie.ts]: Use `encodeURIComponent`

**Review by artalar — CHANGES_REQUESTED** [review]

File: packages/persist-web-storage/src/reatomPersistCookie.ts (hunk 2 above)
Comment (Laniman) [reply packages/persist-web-storage/src/reatomPersistCookie.ts]: fixed

File: packages/persist-web-storage/src/reatomPersistCookie.ts (hunk 1 above)
Comment (Laniman) [reply packages/persist-web-storage/src/reatomPersistCookie.ts]: fixed

**Review by artalar — APPROVED** [review]

**Pull request merged**